tools/scripts/checkCommitMsg.ts

```diff
@@ -0,0 +1,39 @@
+import * as fs from 'fs';
+
+let msgPath = process.argv[2]
+let msg = fs.readFileSync(msgPath, 'utf-8')
+
+let msgArr = msg.split(' ')
+let scope = msgArr[0]
+let project = msgArr[1]
+let issue = msgArr[msgArr.length - 1]
+  .replace(/\n/, ``)
+let summary = msg.replace(scope, ` `)
+  .replace(project, ``)
+  .replace(issue, ``)
+  .replace(/ /g, ``)
+let isValid = true
+
+if(!/:.+:/.test(scope)) {
```

Comment:
In order to avoid too much mental overhead, I'd like to restrict this to a subset of [this emoji list](https://gist.github.com/parmentf/035de27d6ed1dce0b36a) or https://gitmoji.dev/

For example:

- Release/tag - :bookmark: `:bookmark:`
- Feature - :sparkles: `:sparkles:`
- Docs - :books: `:books:`
- Bugfix - :bug: `:bug:`
- Testing - :white_check_mark: `:white_check_mark:`
- Lint/format - :art: `:art:`
- Refactor - :hammer: `:hammer:`
- Code/file removal - :fire: `:fire:`
- CI/CD - :green_heart: `:green_heart:`
- Deps - :lock: `:lock:`
- Breaking changes -  :boom: `:boom:`
- Config - :wrench: `:wrench:`

Let me know if you have others you'd like to include from the start

Reply:
Hmm...I'm not sure we should limit it into a subset at this time, because the `gitmoji` list will keep update. Maybe we can open an issue about the contribute rule to discuss it.

Reply:
No problem, we don't need to limit it right now. I'll keep an issue open with suggested emojis, and we can add a restriction later once it's stable.